L3/CustomViews/TextFields/CardTextField.swift

```diff
@@ -0,0 +1,278 @@
+    
+    enum Const {
+        static let padding: CGFloat = 10
+    }
+    
+    // colors
+    private var defaultTextColor: UIColor = .black
+    private var defaultBorderColor: UIColor = .lightGray
+    private var defaultUnderlineColor: UIColor = .green
+    private var defaultCounterColor: UIColor = .black
+    private var defaultSwitcherColor: UIColor = .green
+    private var defaultPlacehonderColor: UIColor = .lightGray
+    private var defaultTitleColor: UIColor = .lightGray
+    
+    var textField = BaseTextField()
+    private var counter = UILabel()
+    private var bottomBorder = UIView()
+    private var switcher = UISwitch()
+    private var bottomBorderHeightConstraintBold: NSLayoutConstraint?
+    private var bottomBorderHeightConstraintLight: NSLayoutConstraint?
+    private var errorMessageHeightAnchor: NSLayoutConstraint?
+    private var titleLabel: UILabel = {
+        let titleLabel = UILabel()
+        titleLabel.text = " Card number "
+        titleLabel.textColor = .gray
+        titleLabel.font = UIFont.systemFont(ofSize: 12)
+        titleLabel.isHidden = true
+        
+        return titleLabel
+    }()
+    private var errorMessage: UILabel = {
+        let errorMessage = UILabel()
+        errorMessage.text = "Card number should be 16 digits long."
+        errorMessage.textColor = .red
+        errorMessage.font = UIFont.systemFont(ofSize: 14)
+        errorMessage.isHidden = true
+        
+        return errorMessage
+    }()
+    private var textFieldBorderColor: UIColor = UIColor.lightGray {
+        didSet {
+            setNeedsDisplay()
+        }
+    }
+    
+    override init(frame: CGRect) {
+        super.init(frame: frame)
+        textField.delegate = self
+        configureView()
+        configureUI()
+        configureLayout()
+        switcherSetup()
+        configureToolBar()
+    }
+    
+    required init?(coder: NSCoder) {
+        super.init(coder: coder)
+    }
+    
+    override func draw(_ rect: CGRect) {
+        textFieldBorderColor.set()
+        UIBezierPath(roundedRect: CGRectInset(self.bounds, 0.5, 0.5), cornerRadius: 10).stroke()
+    }
+    
+    private func configureView() {
+        backgroundColor = .systemBackground
+        
+        titleLabel.backgroundColor = .systemBackground
+        
+        addSubview(textField)
+        addSubview(bottomBorder)
+        addSubview(counter)
+        addSubview(errorMessage)
+        addSubview(titleLabel)
+    }
+    
+    private func configureUI() {
+        bottomBorder.backgroundColor = .systemGreen
+        
+        counter.text = "Count: 0"
+        counter.font = UIFont.systemFont(ofSize: 12)
+        
+        textField.leftView = switcher
+        textField.leftViewMode = .always
+    }
+    
+    private func switcherSetup() {
+        switcher.addTarget(self, action: #selector(switcherValueChanged), for: .valueChanged)
+    }
+    
+    @objc private func switcherValueChanged() {
+        if switcher.isOn == true {
```

Comment:
```suggestion
        if switcher.isOn {
```

would be enough, since it is bool by itself